test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -2093,16 +2089,4 @@ public void testExpressionResultReferenceWontPreventCollapse() {
         + "ns$Outer;\n"
         + "var ns$Outer$Inner=function(){};\n");
   }
-
-  public void testNoCollapseWithInvalidEnums() {
-    test("/** @enum { { a: { b: number}} } */"
-        + "var e = { KEY1: { a: { /** @nocollapse */ b: 123}},\n"
-        + "  KEY2: { a: { b: 456}}\n"
-        + "}", null, null, CollapseProperties.INVALID_NOCOLLAPSE);
-
-    test("/** @enum */ var e = { A: 1, B: 2 };\n"
-        + "/** @type {{ c: { d: number } }} */ e.name1 = {"
-        + "  c: { /** @nocollapse */ d: 123 } };", null, null,
-        CollapseProperties.INVALID_NOCOLLAPSE);
-  }
 }
```

Comment:
I'm not following. #963 is about allowing nocollapse on enum properties outside the object literal. Does this PR handle that? For example, will it respect nocollapse in this code?

    /** @enum */
    var Foo = {A: 1}
    
    /** @nocollapse */
    Foo.bar = 2;

If not, then we should not merge it, and we should keep the warning about nocollapse on enums. It would be confusing to support nocollapse on some, but not all enum properties.